api/src/main/java/org/openmrs/module/attachments/AttachmentsServiceImpl.java

```diff
@@ -141,15 +137,13 @@
 	@Transactional
 	@Override
 	public Attachment save(Attachment delegate, String reason) {
-		FlushMode flushMode = DbSessionUtil.getCurrentFlushMode();
-		DbSessionUtil.setManualFlushMode();
 		Attachment attachment = new Attachment();
```

Comment:
I understand getting rid of the class from the emrapi module. But how about the functionality of disabling automatic hibernate flushing?

Reply:
I could effectively use the [DbSessionFactory](https://github.com/openmrs/openmrs-core/blob/b7c03a28c64493023a8211c53bb60d4d944b39b7/api/src/main/java/org/openmrs/api/db/hibernate/DbSessionFactory.java) but hesitated seeing no implication in removing the functionality. Also this seems hacky it would imply having dao functionalities at the service layer. Maybe I'm missing a point you could highlight clearly @dkayiwa.

Reply:
Did you confirm that you can successfully save an attachment via this method with your new changes, without any errors?

Reply:
Yes @dkayiwa.